src/d_main.c

```diff
@@ -2743,17 +2743,18 @@ void D_DoomMain(void)
       // frame syncronous IO operations
       I_StartFrame ();
 
-      TryRunTics (); // will run at least one tic
+      if (TryRunTics()) // will run at least one tic
+      {
+        // killough 3/16/98: change consoleplayer to displayplayer
+        S_UpdateSounds(players[displayplayer].mo);// move positional sounds
```

Comment:
How about moving `S_UpdateSounds()` to `TryRunTics()`?

Also do we really need `I_UpdateSound()`? From my testing it seems not.

Reply:
> How about moving S_UpdateSounds() to TryRunTics()?

Sure I can do that.

> Also do we really need I_UpdateSound()? From my testing it seems not.

I don't think so. The function is from the original Doom source release so I left it alone. However, the body of the function was changed for WinMBF and at some point has lost meaning. I suggest either removing it or changing it to a NOP like `I_StartFrame()`.

Reply:
Let's just remove it. After a rewrite with such a footprint, I don't think it makes sense to maintain legacy dummy functions just for that retro feeling anymore. 😉

Reply:
> Let's just remove it.

Done: https://github.com/fabiangreffrath/woof/pull/1118/commits/78ad4e6f4c3898807a79da89c9a66f3702e3ecde

Reply:
> How about moving S_UpdateSounds() to TryRunTics()?

I can't think of a way of doing this without using `goto`. Roughly like this:

```diff
+diff --git a/src/d_loop.c b/src/d_loop.c
index 83de1f2..9b83a8b 100644
--- a/src/d_loop.c
+++ b/src/d_loop.c
@@ -817,7 +817,7 @@ boolean TryRunTics (void)
             // forever - give the menu a chance to work.
             if (I_GetTime() / ticdup - entertic >= MAX_NETGAME_STALL_TICS)
             {
-                return true;
+                goto end;
             }
 
             I_Sleep(1);
@@ -831,7 +831,7 @@ boolean TryRunTics (void)
 
         if (!PlayersInGame())
         {
-            return true;
+            goto end;
         }
 
         set = &ticdata[(gametic / ticdup) % BACKUPTICS];
@@ -859,5 +859,6 @@ boolean TryRunTics (void)
 	NetUpdate ();	// check for new console commands
     }
 
-    return true;
+end:
+    S_UpdateSounds(players[displayplayer].mo);
 }
```
If `goto` is okay, I'll do something like that.

Reply:
Okay, `I_UpdateSound()` is still needed because `i_sound.c` is responsible for updating the `channelinfo[channel].enabled` flag as needed. I suggest no changes to this or `TryRunTics()`.

Reply:
I found the culprit: https://github.com/fabiangreffrath/woof/blob/06939bd9a6501e99ae55e27a5879df6512d10d33/src/s_sound.c#L109-L116
If that `I_SoundIsPlaying()` check is removed, I believe `I_UpdateSound()` is no longer needed. Let me work through this a bit more.

Reply:
Okay, we're good now.

> How about moving S_UpdateSounds() to TryRunTics()?

https://github.com/fabiangreffrath/woof/pull/1118/commits/80da43622a1486c4072e5c5b16a6a5d27083dfb2

> Also do we really need I_UpdateSound()? From my testing it seems not.
> Let's just remove it.

https://github.com/fabiangreffrath/woof/pull/1118/commits/9b096ae11e0fd090c19de4ef58ba9065aded8261